### PR TITLE
Add contributing guidelines

### LINKED ISSUE #124: Create contributing guidelines for Paricia
We will create clear contributing guidelines to further develop Paricia, focusing on good software engineering practices and using quality assurance tooling.

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
Include contributing guidelines in the `CONTRIBUTING.md` file.

Fixes #124

### CODE REVIEW

**Review by dalonsoa — CHANGES_REQUESTED** [review]
This is a good starting point, although I was thinking on including more detailed instructions on the different ways people could contribute). The thing I had in mind is something like this - although adapted to Paricia, of course:

https://github.com/qpv-research-group/solcore5/blob/develop/CONTRIBUTING.md

In particular, it might need to include some instructions on how to organise a "developers's setup" using Docker and pre-commit hooks, for example. They are already included in the README, but I feel this might be a better place than the README, to be honest:

- https://github.com/ImperialCollegeLondon/paricia?tab=readme-ov-file#getting-started
- https://github.com/ImperialCollegeLondon/paricia?tab=readme-ov-file#tests
- https://github.com/ImperialCollegeLondon/paricia?tab=readme-ov-file#continuous-integration

The README needs updating, anyway, as some sections there are obsolete.

**Comment by SaranjeetKaur** [comment]
Looks like there is no `CODE OF CONDUCT` file on the repo and not sure whom to report, just in case. I have include a CoC section on line 32, let me know if it needs to be removed or.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+[Developer's setup](#developers-setup)
+
+- [Installing Paricia](#installing-paricia)
+- [Tests](#tests)
+- [Continuous integration](#continuous-integration)
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by the [Paricia Code of Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [the repository Administrator](mailto:).
```
Comment (dalonsoa) [inline CONTRIBUTING.md:34]: I've added a generic CoC in #255 , merging into this branch. About the contact, I've used Wouter's email there, and probably the same should go in here. @ICHydro , could you confirm you're happy to appear here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+- **Did the problem start happening recently** (e.g. after updating to a new version of Paricia) or was this always a problem?
+- If the problem started happening recently, **can you reproduce the problem in an older version of Paricia?** What's the most recent version in which the problem doesn't happen? You can download older versions of Paricia from [the releases page](https://github.com/ImperialCollegeLondon/paricia/releases).
+- **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+Include details about your configuration and environment:
+
+- **Which version of Paricia are you using?**
+- **What's the name and version of the OS you're using**?
+- **Are you running Paricia in a virtual machine?** If so, which VM software are you using and which operating systems and versions are used for the host and the guest?
+- **If the problem is related with the Poisson-drift-diffusion solver**, which compiler did you use? What were the installation steps you followed?
```
Comment (dalonsoa) [inline CONTRIBUTING.md:68]:
```suggestion
```

**Review by dalonsoa — APPROVED** [review]
I've made a couple of comments but otherwise this looks great! Excellent work!

**Comment by dalonsoa** [comment]
@ICHydro , do you have any comment on this before we merge this? This contributing guidelines is what researchers (here and there) will follow, so it would be good to have your feedback before making this formal.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ICHydro) [reply CONTRIBUTING.md:34]: Sure. You can also link to my Imperial profile page, but email is file.

**Review by ICHydro — APPROVED** [review]
Looks all good to me!

**Pull request merged**